keps/sig-autoscaling/2702-graduate-v2beta2-autoscaling-api-to-GA/README.md

```diff
@@ -0,0 +1,422 @@
+
+### Non-Goals
+
+* Promote scale-to-zero feature as a part of this effort. Since it requires additional effort
+  to deprecate the special flag meaning of scaling subresource replicas=0 (disable autoscaling).
+  [Progress](https://github.com/kubernetes/enhancements/issues/2021) has been made. However,
+  it is not part of HPA v2 stable effort since APIs cannot introduce breaking changes.
+  
+
+## Proposal
+
+
+### Implementation Details
+
+
+### Risks and Mitigations
+
+* v1-v2 [conversion loss](https://github.com/kubernetes/kubernetes/issues/80481) of multiple CPU targets
+* v2beta1 has the significant amount of boilerplate and overhead in maintaining conversion routines for multiple public APIs.
+* Avoid existing confusion with `SelectPolicy` [enumeration](https://github.com/kubernetes/api/blob/2c3c141c931c0ab1ce1396c3152c72852b3d37ee/autoscaling/v2beta2/types.go#L149-L156)
+  by replacing `Min` and `Max` with respective `MinChange` and `MaxChange` in v2 stable.
+* Replace the value `Disabled` with `ScalingDisabled` for better [understanding](https://github.com/kubernetes/kubernetes/pull/95647#discussion_r507563282)
+
+## Design Details
+
+### Test Plan
+
+* Add e2e tests for HPA behavior
+* The KEP [test plan](https://github.com/kubernetes/enhancements/blob/15d330a932e0aae220ff719c391f7d815492088f/keps/sig-autoscaling/20190307-configurable-scale-velocity-for-hpa.md#test-plan) includes unit tests
+* Add e2e tests for container resource targets
+
+### Graduation Criteria
+
+The following code changes must be made for graduating to GA
+
+* Move API objects to `v2` and support conversion internally
+
+The following code changes must be made to take <TBD> GA
+
+* TBD
+
+### Version Skew Strategy
+
+### Upgrade/Downgrade Strategy
+
+## Production Readiness Review
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
```

Comment:
yeah, this is going to stable so it's no longer applicable. if there is an existing feature gate though, it would be good to document it so that folks seeing this documentation when on older versions will know what to do.